### DISCUSSION TITLE
Bun v1.0.3

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#5906] [Announcements]
To install Bun v1.0.3:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade to Bun v1.0.3:
```bash
bun upgrade
```

**[Read Bun v1.0.3's release notes on Bun's blog](https://bun.sh/blog/bun-v1.0.3)**

## What's Changed
* VSCode Extension Bunlock Syntax-Highlighting and Docs Improvements by @JeremyFunk in https://github.com/oven-sh/bun/pull/4652
* docs(runtime): fix some typo. by @zongzi531 in https://github.com/oven-sh/bun/pull/5451
* Update development.md by @sonyarianto in https://github.com/oven-sh/bun/pull/5531
* fix(corking) uncork if needed by @cirospaciari in https://github.com/oven-sh/bun/pull/5525
* fix(request) handle undefined/null/empty signal on request by @cirospaciari in https://github.com/oven-sh/bun/pull/5503
* fix(bundler): Add a space before minified require by @davidmhewitt in https://github.com/oven-sh/bun/pull/5521
* fix(node/fs.watch): Check first char before trimming event filenames by @davidmhewitt in https://github.com/oven-sh/bun/pull/5505
* webkit upgrade by @dylan-conway in https://github.com/oven-sh/bun/pull/5535
* Follow-up for workspace docs #5379 and #5229 by @bdenham in https://github.com/oven-sh/bun/pull/5555
* fix: `ArrayBufferConstructor` type signature by @52 in https://github.com/oven-sh/bun/pull/5579
* fix: `array-buffer.test-d.ts` test by @52 in https://github.com/oven-sh/bun/pull/5580
* fix: node compatibility with empty path string (stat, exist, ...) by @coratgerl in https://github.com/oven-sh/bun/pull/4693
* Make error message when importing a node: module in a browser bundle clearer by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5603
* docs: fix typo in lockflie nav by @Vilsol in https://github.com/oven-sh/bun/pull/5576
* Docs: path aliases fix by @e253 in https://github.com/oven-sh/bun/pull/5593
* fix(fetch) handle 100 continue by @cirospaciari in https://github.com/oven-sh/bun/pull/5496
* feat(encoding): TextDecoder support undefined by @WingLim in https://github.com/oven-sh/bun/pull/5387
* fix(child_process) unref next tick so exit/close event can be fired before application exits by @cirospaciari in https://github.com/oven-sh/bun/pull/5481
* Implement VSCode tasks for bun by @JeremyFunk in https://github.com/oven-sh/bun/pull/5529
* fix(install): Return NotSupported when errno == XDEV by @pan93412 in https://github.com/oven-sh/bun/pull/5610
* Fix ZLS commit hash in the document by @shinichy in https://github.com/oven-sh/bun/pull/5510
* Added .DS_Store to gitignore-for-init by @Cilooth in https://github.com/oven-sh/bun/pull/5628
* Workaround #5604 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5615
* Fix a TypeError in the documentation by @LapsTimeOFF in https://github.com/oven-sh/bun/pull/5626
* Add a way to disable the GC timer by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5656
* Remove hardcoded references to zig in Makefile by @xbjfk in https://github.com/oven-sh/bun/pull/5660
* feat(console.log): Print anonymous when class name is unknown by @JibranKalia in https://github.com/oven-sh/bun/pull/5595
* feat(test): Implement `arrayContaining` by @WingLim in https://github.com/oven-sh/bun/pull/5572
* In `bun:sqlite`, make sure we set the number tag correctly when creating the JSValue by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5655
* fix(config): support for registry url without trailing slash by @Hanaasagi in https://github.com/oven-sh/bun/pull/5662
* fix(docs): update formatting by @rauny-brandao in https://github.com/oven-sh/bun/pull/5685
* docs: add missing options from bun init by @jumoog in https://github.com/oven-sh/bun/pull/5638
* change circles for color-blinds by @Hamcker in https://github.com/oven-sh/bun/pull/5594
* Fix HTTP listen behavior being non-compliant with node by @paperdave in https://github.com/oven-sh/bun/pull/5689
* feat(runtime): Implement `console.Console` by @paperdave in https://github.com/oven-sh/bun/pull/5448
* Implement `node_api_create_external_string_latin1`  by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5675
* fix(runtime/node): Allow `new Buffer.alloc()` + Upgrade WebKit by @paperdave in https://github.com/oven-sh/bun/pull/5699
* fix: remove unneeded branch in toJSONWithBytes by @liz3 in https://github.com/oven-sh/bun/pull/5684
* update llvm version from 15 to 16 in makefile by @nithinkjoy-tech in https://github.com/oven-sh/bun/pull/5696
* fix: provide empty string to 0 length process environment variables by @liz3 in https://github.com/oven-sh/bun/pull/5679
* `bun run` fix missing script error on empty file by @Parzival-3141 in https://github.com/oven-sh/bun/pull/5025
* Add navigator type definition by @ruihe774 in https://github.com/oven-sh/bun/pull/5444
* Encode slashes in package names in the registry manifest request by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5716
* Make bun install --verbose more verbose by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5726
* Fixes #3712 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5730
* Align fetch() redirect behavior with spec by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5729
* Get artifactory to work by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5744
* docs: Update Remix guide by @brookslybrand in https://github.com/oven-sh/bun/pull/5702
* Added react installation to react.md by @jt3k in https://github.com/oven-sh/bun/pull/5620
* remind users of the latest version by @jumoog in https://github.com/oven-sh/bun/pull/5597
* docs: update `net` node documentation by @weyert in https://github.com/oven-sh/bun/pull/5562
* docs(development): typo which would lead to wrong llvm installation by @sum117 in https://github.com/oven-sh/bun/pull/5513
* Doc updates by @colinhacks in https://github.com/oven-sh/bun/pull/5759
* fix(cli): `bun pm cache rm` command not work by @WingLim in https://github.com/oven-sh/bun/pull/4571
* fix(Makefile) by @cirospaciari in https://github.com/oven-sh/bun/pull/5762
* Fixes #5769 by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5775
* add warning to Ensure correct placement of the '--watch' flag by @a4addel in https://github.com/oven-sh/bun/pull/5447
* Updated modules.md to address issue #5420 by @h2210316651 in https://github.com/oven-sh/bun/pull/5456
* docs: add Qwik guide by @sanyamkamat in https://github.com/oven-sh/bun/pull/4810
* Show when a newer version is available in the install summary by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5780
* Fix RedirectURLTooLong errors by @ggobbe in https://github.com/oven-sh/bun/pull/5786
* Fix path used in bunx by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5785
* Call `Error.prepareStackTrace` on `new Error().stack` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5802
* fix(doc): correct `server.reload` by @Hanaasagi in https://github.com/oven-sh/bun/pull/5799
* add `emitDecoratorMetadata` by @dylan-conway in https://github.com/oven-sh/bun/pull/5777
* Fix bugs for VSCode extension 0.0.9 release by @JeremyFunk in https://github.com/oven-sh/bun/pull/5772
* Treat `undefined` value as empty in expect.toThrow by @Brooooooklyn in https://github.com/oven-sh/bun/pull/5788
* feat: switch disableTelemetry to bunfig by @coratgerl in https://github.com/oven-sh/bun/pull/5690
* [bun install] Add `-E` as alias of `--exact` by @jonahsnider in https://github.com/oven-sh/bun/pull/5104
* Improve types for `test.each`, `describe.each` by @colinhacks in https://github.com/oven-sh/bun/pull/5838
* feat(runtime): implement `console._stdout` by @paperdave in https://github.com/oven-sh/bun/pull/5842
* feat(runtime): implement `Module.prototype._compile` [fix `jsonpath`] by @paperdave in https://github.com/oven-sh/bun/pull/5840
* feat(node:dns): implement `dns.lookupService` by @Hanaasagi in https://github.com/oven-sh/bun/pull/5613
* feat(encoding): support BOM detection by @WingLim in https://github.com/oven-sh/bun/pull/5550
* fix(ffi) fix size limit for dlopen by @cirospaciari in https://github.com/oven-sh/bun/pull/5516
* fix(run): interpret extensionless files as typescript by @paperdave in https://github.com/oven-sh/bun/pull/5711
* docs(project): development fix Arch install-dependencies command by @yadav-saurabh in https://github.com/oven-sh/bun/pull/5270
* fix(fetch): fix redirect in relative path location. by @Hanaasagi in https://github.com/oven-sh/bun/pull/5781
* fix(console.log): change default depth from 8 to 2 by @JibranKalia in https://github.com/oven-sh/bun/pull/5839
* fix: add check to sqlite extension loading logic by @liz3 in https://github.com/oven-sh/bun/pull/5773
* Fix `make jsc` on Linux by @AaronDewes in https://github.com/oven-sh/bun/pull/4779
* On Linux, respect memory limit from cgroups by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5849
* Fix test failures by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5862
* chore: Update ZSH completions (1.0.0+822a00c build) by @Hanaasagi in https://github.com/oven-sh/bun/pull/4428
* fix(node:fs): use the right `copyFile` constants by @paperdave in https://github.com/oven-sh/bun/pull/5874
* fix: correctly pass the encrypted(bool) property on "Socket" for express.js by @liz3 in https://github.com/oven-sh/bun/pull/5878
* chore(git): ignore .envrc file by @igorshapiro in https://github.com/oven-sh/bun/pull/5873
* Don't use arena in node:fs by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/5863
* docs: add await to all Bun.build() call by @ImBIOS in https://github.com/oven-sh/bun/pull/5885
* async fs Write deinit and ref buffer by @dylan-conway in https://github.com/oven-sh/bun/pull/5883
* Hide env elapsed time when loglevel=info by @colinhacks in https://github.com/oven-sh/bun/pull/5882
* disable child_process execArgv + fix crash in cli parsing by @paperdave in https://github.com/oven-sh/bun/pull/5843
* fix #5865 by @dylan-conway in https://github.com/oven-sh/bun/pull/5890
* Implement module.parent by @paperdave in https://github.com/oven-sh/bun/pull/5889

## New Contributors
* @JeremyFunk made their first contribution in https://github.com/oven-sh/bun/pull/4652
* @bdenham made their first contribution in https://github.com/oven-sh/bun/pull/5555
* @52 made their first contribution in https://github.com/oven-sh/bun/pull/5579
* @coratgerl made their first contribution in https://github.com/oven-sh/bun/pull/4693
* @Vilsol made their first contribution in https://github.com/oven-sh/bun/pull/5576
* @pan93412 made their first contribution in https://github.com/oven-sh/bun/pull/5610
* @shinichy made their first contribution in https://github.com/oven-sh/bun/pull/5510
* @Cilooth made their first contribution in https://github.com/oven-sh/bun/pull/5628
* @xbjfk made their first contribution in https://github.com/oven-sh/bun/pull/5660
* @JibranKalia made their first contribution in https://github.com/oven-sh/bun/pull/5595
* @rauny-brandao made their first contribution in https://github.com/oven-sh/bun/pull/5685
* @jumoog made their first contribution in https://github.com/oven-sh/bun/pull/5638
* @Hamcker made their first contribution in https://github.com/oven-sh/bun/pull/5594
* @liz3 made their first contribution in https://github.com/oven-sh/bun/pull/5684
* @nithinkjoy-tech made their first contribution in https://github.com/oven-sh/bun/pull/5696
* @ruihe774 made their first contribution in https://github.com/oven-sh/bun/pull/5444
* @brookslybrand made their first contribution in https://github.com/oven-sh/bun/pull/5702
* @jt3k made their first contribution in https://github.com/oven-sh/bun/pull/5620
* @weyert made their first contribution in https://github.com/oven-sh/bun/pull/5562
* @sum117 made their first contribution in https://github.com/oven-sh/bun/pull/5513
* @a4addel made their first contribution in https://github.com/oven-sh/bun/pull/5447
* @h2210316651 made their first contribution in https://github.com/oven-sh/bun/pull/5456
* @sanyamkamat made their first contribution in https://github.com/oven-sh/bun/pull/4810
* @ggobbe made their first contribution in https://github.com/oven-sh/bun/pull/5786
* @Brooooooklyn made their first contribution in https://github.com/oven-sh/bun/pull/5788
* @jonahsnider made their first contribution in https://github.com/oven-sh/bun/pull/5104
* @yadav-saurabh made their first contribution in https://github.com/oven-sh/bun/pull/5270
* @AaronDewes made their first contribution in https://github.com/oven-sh/bun/pull/4779
* @igorshapiro made their first contribution in https://github.com/oven-sh/bun/pull/5873
* @ImBIOS made their first contribution in https://github.com/oven-sh/bun/pull/5885

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v1.0.2...bun-v1.0.3

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v1.0.3'>Bun v1.0.3</a>.</em>

**Comment by marcelloinfoweb** [comment]
The changelog could be divided: New, fixed, removed, etc...

**Reply by jumoog** [reply]
yes using https://github.com/conventional-changelog/commitlint and https://github.com/conventional-changelog/releaser-tools/tree/master/packages/conventional-github-releaser
